### ISSUE TITLE
Job callbacks aren't executed when Queue is initialized with `is_async=False`

### ISSUE THREAD

**Issue by lewin** [rq/rq#1586]
Small example
```
# somewhere.py
def say_hello():
    print("hello")
def success_callback(job, result, connection):
    print("callback")

# main.py
from somewhere import say_hello, success_callback
from fakeredis import FakeStrictRedis
from rq import Queue, SimpleWorker

queue = Queue(is_async=False, connection=FakeStrictRedis())
# this should print "hello" and "callback", only prints "hello"
queue.enqueue(say_hello, on_success=success_callback)

queue = Queue(connection=FakeStrictRedis())
worker = SimpleWorker([queue], connection=queue.connection)
queue.enqueue(say_hello, on_success=success_callback)
# this works, it prints both
worker.work(burst=True)
```

**Comment by selwin** [comment]
Mind opening a PR for this?

**Comment by ericman93** [comment]
the issue with the simple worker solution is that you can't use mocks, since the mocks are being serialized to the worker and when asserting, its not the same mock that was called

**Comment by tomhamiltonstubber** [comment]
This is fixed now, right?

**Comment by lowercase00** [comment]
Should be fixed with #1599. Feel free to reopen if was not.
